installer.sh

```diff
@@ -0,0 +1,442 @@
+        output "$line" "comment"
+    done
+    IFS=$OLDIFS
+}
+
+# Check that cURL is installed
+function check_curl() {
+    if command -v curl >/dev/null 2>&1; then
+        output "  [*] cURL is installed" "success"
+        if gh auth status >/dev/null 2>&1; then
+            GITHUB_TOKEN="$(gh auth token)"
+            if ! github_curl https://api.github.com/repos/platformsh/cli/releases/latest >/dev/null 2>&1; then
+                GITHUB_TOKEN=""
+            else
+                output "  [*] Using GitHub auth from the gh CLI" "success"
+            fi
+        elif [ ! -z "${GITHUB_TOKEN}" ]; then
+            if ! github_curl https://api.github.com/repos/platformsh/cli/releases/latest >/dev/null 2>&1; then
+                GITHUB_TOKEN=""
+            else
+                output "  [*] Using GitHub auth from the GITHUB_TOKEN env variable" "success"
+            fi
+        fi
+    else
+        output "  [ ] ERROR: cURL is required for installation" "error"
+        exit_with_error
+    fi
+}
+
+# Check that Gzip is installed
+function check_gzip() {
+    if command -v gzip >/dev/null 2>&1; then
+        output "  [*] Gzip is installed" "success"
+    else
+        output "  [ ] ERROR: Gzip is required for installation" "error"
+        exit_with_error
+    fi
+}
+
+function check_version() {
+    if [ -z "${VERSION}" ]; then
+        version=$(github_curl https://api.github.com/repos/platformsh/cli/releases/latest | grep "tag_name" | cut -d \" -f 4)
+        output "  [*] No version specified, using latest ($version)" "success"
+    else
+        output "  [*] Version ${VERSION} specified" "success"
+        version=${VERSION}
+    fi
+}
+
+# Detect the kernel type
+function check_kernel() {
+    kernel=$(uname -s 2>$pipe || /usr/bin/uname -s)
+    case ${kernel} in
+        "Linux"|"linux")
+            kernel="linux"
+            ;;
+        "Darwin"|"darwin")
+            kernel="darwin"
+            dir_bin="/usr/local/bin"
+            ;;
+        "FreeBSD"|"freebsd")
+            kernel="freebsd"
+            ;;
+        *)
+            output "  [ ] Your OS (${kernel}) is currently not supported" "error"
+            exit_with_error
+            ;;
+    esac
+
+    output "  [*] Your kernel (${kernel}) is supported" "success"
+}
+
+function check_architecture() {
+    # Detect architecture
+    machine=$(uname -m 2>$pipe || /usr/bin/uname -m)
+    case ${machine} in
+        aarch64*|armv8*|arm64*)
+            machine="arm64"
+            ;;
+        i[36]86|x86)
+            machine="386"
+            ;;
+        x86_64|amd64)
+            machine="amd64"
+            ;;
+        *)
+            output "  [ ] Your architecture (${machine}) is currently not supported" "error"
+            exit_with_error
+            ;;
+    esac
+
+    output "  [*] Your architecture (${machine}) is supported" "success"
+}
+
+function check_install_method() {
+    if [ "linux" == $kernel ]; then
```

Comment:
```suggestion
    if [ "$kernel" = linux ]; then
```
etc.

there are a few `if` statements where it would be safer to double-quote the variable (everywhere there is a variable) to avoid problems with special chars

```
bash-3.2$ export GITHUB_TOKEN='foo bar'
bash-3.2$ if [ -z $GITHUB_TOKEN ] ; then echo empty ; fi
bash: [: foo: binary operator expected
```

Reply:
Good catch, even if invalid we should handle this indeed 👍🏽